### ISSUE TITLE
Specific filter does not give correct results

### ISSUE THREAD

**Issue by someren** [Signalen/backend#149]
One specific filter does not give the correct results. It either shows all signals or shows the results from the previous filter.
I was unable to determine a cause.
 These are my conclusions so far:
- Go to "Mijn filters" and click "Toon resultaat" does not work, you either get all signals as a result or the results from a previous filter.
- Rebuilding this filter with "Filteren" and not saving it will result in correctly shown signals. Therefore it seems that the filter itself works but something goes wrong after saving and displaying the results?
- Building and saving this filter step by step works fine. For example: Status + Categories, Categories + Location, Location + Afdeling, Afdeling + Status, Categories + Location + Afdeling, Status + Location + Categories etc. They all work, even when opening the results from a saved filter. 
- I can reproduce the issue with other, similar filters that have a combination of: Status, Location, Categorie and Afdeling. After saving and clicking on "Toon resultaat" I get all signals or the results from the previous filter. Building the filter from scratch and not saving it does give the correct results.

Filter with incorrect results:
![image](https://user-images.githubusercontent.com/66589775/114350068-ff79ab80-9b68-11eb-938a-98015a6d129d.png)

Could you please find out why this filter, the combination of chosen fields or the save and "Toon results" option does not work properly?

**Comment by someren** [comment]
Closed the issue, it cannot be reproduced on version 1.45.

**Comment by someren** [comment]
Reopend. New version: BE 2.0.0 FE 1.46. This specific filter does not work anymore. Last time (v2.0.0 1.45) I did get correct results... Weird that this specific filter does not work properly. Note: when opening from saved filters. Creating the filter one time does give correct results.

**Comment by someren** [comment]
Another example of a different filter that does not give correct results, either everything or the results of the previous filter:

![Filter Maashoeve](https://user-images.githubusercontent.com/66589775/117805318-1dfebf80-b259-11eb-864b-8f8e558deb23.PNG)

**Comment by someren** [comment]
Another example of a filter which does not give correct results:
![Filter Chantal](https://user-images.githubusercontent.com/66589775/120595921-67e75980-c443-11eb-8ee4-7e87adabe863.PNG)

Results show area's which are not selected:
![Filter Chantal resultaat](https://user-images.githubusercontent.com/66589775/120595950-703f9480-c443-11eb-842e-94d1fc8a3af1.PNG)

**Comment by someren** [comment]
@boris-arkenaar we discussed filters with incorrect results last tuesday. In this ticket you will find several examples.

**Comment by someren** [comment]
Issue is still present. I still do not know the exact cause. User applied filter on multiple Subcategory, Area, Status and Medewerker=Niet toegewezen. The result is either all signals or the results from the previous filter are shown.

**Comment by boris-arkenaar** [comment]
I can't directly reproduce this. Does it only happen when you select so many options per Subcategory, Area, Status?

**Comment by someren** [comment]
Yes, only the very extensive/large filters have problems. Note: the result is correct when you make an incidental filter. Only after opening a saved filter the results are incorrect.

**Comment by someren** [comment]
Additional information: Most likely, the filter/search request is too large. There is no error or warning message that informs the user about the incorrect result.
